R/estimatr_lh_robust.R

```diff
@@ -0,0 +1,184 @@
+#' Linear Hypothesis for Ordinary Least Squares with Robust Standard Errors
+#'
+#' @description This function is a wrapper for \code{\link{lm_robust}} that
+#' is useful for estimating linear combination of coefficients.
+#'
+#' @param formula an object of class formula, as in \code{\link{lm}}
+#' @param data A \code{data.frame}
+#' @param weights the bare (unquoted) names of the weights variable in the
+#' supplied data.
+#' @param subset An optional bare (unquoted) expression specifying a subset
+#' of observations to be used.
+#' @param clusters An optional bare (unquoted) name of the variable that
+#' corresponds to the clusters in the data.
+#' @param fixed_effects An optional right-sided formula containing the fixed
+#' effects that will be projected car_lht of the data, such as \code{~ blockID}. Do not
+#' pass multiple-fixed effects with intersecting groups. Speed gains are greatest for
+#' variables with large numbers of groups and when using "HC1" or "stata" standard errors.
+#' See 'Details'.
+#' @param se_type The sort of standard error sought. If \code{clusters} is
+#' not specified the options are "HC0", "HC1" (or "stata", the equivalent),
+#'  "HC2" (default), "HC3", or
+#' "classical". If \code{clusters} is specified the options are "CR0", "CR2" (default), or "stata". Can also specify "none", which may speed up estimation of the coefficients.
+#' @param ci logical. Whether to compute and return p-values and confidence
+#' intervals, TRUE by default.
+#' @param alpha The significance level, 0.05 by default.
+#' @param return_vcov logical. Whether to return the variance-covariance
+#' matrix for later usage, TRUE by default.
+#' @param try_cholesky logical. Whether to try using a Cholesky
+#' decomposition to solve least squares instead of a QR decomposition,
+#' FALSE by default. Using a Cholesky decomposition may result in speed gains, but should only
+#' be used if users are sure their model is full-rank (i.e., there is no
+#' perfect multi-collinearity)
+#' @param linearHypothesis A character string or a matrix specifying combination.
```

Comment:
Maybe this is a good time to have `lm_lin` and `lh_robust` borrow argument names from `lm_robust` rather than repeating them? Would be good to just have them pull from the same set once for overlapping arguments.

Reply:
+1 for inheritParams - see eg https://github.com/DeclareDesign/DeclareDesign/blob/master/R/aaa.R#L199-L206 for how I did that in DeclareDesign

Reply:
Not doing that exactly.
Adding `...` arg and linking to `lm_robust`